google-cloud-bigtable/test/google/cloud/bigtable/conformance_test.rb

```diff
@@ -51,29 +50,37 @@ def self.build_test_for test, index
                               app_profile_id: nil
                             ]
 
-      results = test.results.to_a
-      if results.map(&:error).include? true
+      rows = table.read_rows
+      if test.results.empty? # "no data after reset"
         expect do
-          table.read_rows.to_a # to_a is needed because Enum lazily invokes read_rows
-        end.must_raise Google::Cloud::Bigtable::InvalidRowStateError
+          rows.next
+        end.must_raise StopIteration
```

Comment:
Is this basically saying "There are no rows"? Is there an implicit assertion that the loop below raise StopIteration, because that's how the "do" loop works? Sorry for what is probably a very basic question - I don't know Ruby at all.

Reply:
Ah, no - I've just thought, you're currently checking in the loop below that every expected result has a matching actual result, but I don't *think* you're checking that when we get to the end of the expected results, we've got to the end of the actual results.

I suspect you need another

```rb
expect do
  rows.next
end.must_raise StopIteration
```

at the end of the loop (line ~84).

Please let me know if I've misunderstood though!

Reply:
Thanks @jskeet, that's a very good idea to check for unexpected cells. Unfortunately, I tried it, and it results in the client attempting another RPC:

```
[StopIteration] exception expected, not
Class: <MockExpectationError>
Message: <"No more expects available for :read_rows: [\"projects/test/instances/test-instance/tables/test-table\", {:rows=><Google::Bigtable::V2::RowSet: row_keys: [], row_ranges: []>, :filter=>nil, :rows_limit=>nil, :app_profile_id=>nil}]">
```

So I think I'll have to leave it out.

Reply:
Looking at it with @blowmage, we managed to add the end of loop check, with a conditional to skip it if an `InvalidRowStateError` has been raised.